src/Core/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommand.cs

```diff
@@ -212,68 +218,43 @@ private async Task SendServiceAccountLimitEmailAsync(Organization organization,
             throw new BadRequestException("Plan does not allow additional Secrets Manager seats.");
         }
 
-        if (plan.BaseSeats > newSeatCount)
+        if (plan.BaseSeats > update.NewTotalSeats)
         {
             throw new BadRequestException($"Plan has a minimum of {plan.BaseSeats} Secrets Manager  seats.");
         }
 
-        if (newSeatCount <= 0)
+        if (update.NewTotalSeats <= 0)
         {
             throw new BadRequestException("You must have at least 1 Secrets Manager seat.");
         }
 
-        var additionalSeats = newSeatCount - plan.BaseSeats;
-
-        if (plan.MaxAdditionalSeats.HasValue && additionalSeats > plan.MaxAdditionalSeats.Value)
+        if (plan.MaxAdditionalSeats.HasValue && update.NewAdditionalSeats > plan.MaxAdditionalSeats.Value)
         {
             throw new BadRequestException($"Organization plan allows a maximum of " +
                                           $"{plan.MaxAdditionalSeats.Value} additional Secrets Manager seats.");
         }
 
-        if (!organization.SmSeats.HasValue || organization.SmSeats.Value > newSeatCount)
+        if (!organization.SmSeats.HasValue || organization.SmSeats.Value > update.NewTotalSeats)
```

Comment:
`!organization.SmSeats.HasValue` is already checked in line 196

Reply:
yeah